pkg/discovery/worker/group_discovery_worker.go

```diff
@@ -255,155 +247,10 @@ func (worker *k8sEntityGroupDiscoveryWorker) BuildTurboPolicyDTOsFromPolicyBindi
-	}
-	var settingValue float32
-	for _, setting := range spec.Objectives {
-		if err := json.Unmarshal(setting.Value.Raw, &settingValue); err != nil {
-			return nil, err
-		}
-		if settingValue <= 0 {
-			return nil, fmt.Errorf("setting %v has a non-positive value %v",
-				setting.Name, settingValue)
-		}
-		switch setting.Name {
-		case v1alpha1.ResponseTime:
-			settings.AddSetting(group.NewResponseTimeSLOPolicySetting(settingValue))
-		case v1alpha1.Transaction:
-			settings.AddSetting(group.NewTransactionSLOPolicySetting(settingValue))
-		default:
-			return nil, fmt.Errorf("unknown objective name %v", setting.Name)
-		}
-	}
-	behavior := spec.Behavior
-	if behavior.HorizontalScaleDown != nil {
-		mode, err := validateActionMode(*behavior.HorizontalScaleDown)
-		if err != nil {
-			return nil, err
-		}
-		settings.AddSetting(group.NewHorizontalScaleDownAutomationPolicySetting(mode))
-	}
-	if behavior.HorizontalScaleUp != nil {
-		mode, err := validateActionMode(*behavior.HorizontalScaleUp)
-		if err != nil {
-			return nil, err
-		}
-		settings.AddSetting(group.NewHorizontalScaleUpAutomationPolicySetting(mode))
-	}
-	displayName := fmt.Sprintf("TurboPolicy::%v on %v [%v]",
-		policyBinding.GetPolicyType(), policyBinding, worker.targetId)
-	return group.NewSettingPolicyBuilder().
-		WithDisplayName(displayName).
-		WithName(policyBinding.GetUID()).
-		WithSettings(settings.Build()).
-		Build()
-}
-
-// validateReplicas validates minReplicas and maxReplicas
-func validateReplicas(minReplicas *int32, maxReplicas *int32) (*int32, *int32, error) {
-	if minReplicas == nil && maxReplicas == nil {
-		return nil, nil, nil
-	}
-	if minReplicas != nil && !isWithinValidRange(*minReplicas) {
-		return nil, nil, fmt.Errorf("invalid minReplicas %v. Must be between 1 and 10000 inclusive", *minReplicas)
-	}
-	if maxReplicas != nil && !isWithinValidRange(*maxReplicas) {
-		return nil, nil, fmt.Errorf("invalid maxReplicas %v. Must be between 1 and 10000 inclusive", *maxReplicas)
-	}
-	if minReplicas != nil && maxReplicas != nil && *minReplicas > *maxReplicas {
-		return nil, nil, fmt.Errorf("minReplicas %v is larger than maxReplicas %v", *minReplicas, *maxReplicas)
-	}
-	return minReplicas, maxReplicas, nil
-}
-
-// isWithinValidRange check if the number of replicas is within the valid range of 1 to 10000
-func isWithinValidRange(replicas int32) bool {
-	if replicas >= 1 && replicas <= 10000 {
-		return true
-	}
-	return false
-}
-
-// validateActionMode validates the action mode
-// Valid action mode must exist in actionModes map
-func validateActionMode(actionMode v1alpha1.ActionMode) (string, error) {
-	if mode, ok := actionModes[actionMode]; ok {
-		return mode, nil
-	}
-	return "", fmt.Errorf("unsupported action mode %v", actionMode)
+func (worker *k8sEntityGroupDiscoveryWorker) buildContainerVerticalScalePolicy(
+	policyBinding *repository.TurboPolicyBinding) (*proto.GroupDTO, error) {
+	return buildContainerVerticalScalePolicy(worker, policyBinding)
```

Comment:
How is 'err' from buildContainerVerticalScalePolicy sent up to caller

Reply:
will try to address this in https://vmturbo.atlassian.net/browse/OM-98560